File: 10-10-2026/Q33.py
import matplotlib.pyplot as plt
import numpy as np

def f(x):
    y = np.exp(x)-2
    return y

x_vals = np.linspace(-2,2,500)
y_vals=f(x_vals)

plt.plot(x_vals, y_vals)
plt.grid(True)

xV=[np.log(2)]
yV=[0]

plt.plot(xV, yV, 'ro')
plt.text(np.log(2), 0, "(ln(2),0)")

plt.savefig('figure33.png')

File: 10-10-2026/Q33.c
#include<stdio.h>
#include<math.h>

float f(float x){
	float y = exp(x)-2;
	return y;
}

float g(float x){
	float z = x - f(x)/(f(x)+2);
	return z;
}

int main(){
	float x0=1;
	float x = g(x0);

	while(x0-x>0.001){
		x0 = x;
		x = g(x);
	}
	
	printf("%f %f",x,f(x));

	return 0;
}

